app/charts/chart-config-ui-options.ts

```diff
@@ -284,8 +288,23 @@ export const chartConfigOptionsUISpec: ChartSpecs = {
         componentTypes: SEGMENT_COMPONENT_TYPES,
         filters: true,
         sorting: AREA_SEGMENT_SORTING,
-        getDisabledState: (_chartConfig, _, data) => {
+        getDisabledState: (_chartConfig, components, data) => {
           const chartConfig = _chartConfig as AreaConfig;
+          const yIri = chartConfig.fields.y.componentIri;
+          const yDimension = components.find((d) => d.iri === yIri);
+          const disabledStacked = disableStacked(yDimension);
+
+          if (disabledStacked) {
+            return {
+              disabled: true,
+              warnMessage: t({
+                id: "controls.segment.stacked.disabled-by-scale-type",
+                message:
```

Comment:
Stacked layout can only be enabled if the vertical axis dimension has a ratio scale.